src/main/java/org/aksw/simba/squirrel/metadata/MetaDataHandler.java

```diff
@@ -35,18 +47,30 @@ public MetaDataHandler(String updateDatasetURI, String queryDatasetURI) {
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.endedAtTime.asNode(), NodeFactory.createLiteral(crawlingActivity.getDateEnded())));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.status.asNode(), NodeFactory.createLiteral(crawlingActivity.getState().toString())));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.wasAssociatedWith.asNode(), NodeFactory.createLiteral(String.valueOf(crawlingActivity.getWorker().getId()))));
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.numberOfTriples.asNode(), NodeFactory.createLiteral(String.valueOf(crawlingActivity.getNumTriples()))));
-        Node nodeResultGraph = NodeFactory.createURI(crawlingActivity.getGraphId());
-        lstTriples.add(new Triple(nodeCrawlingActivity, MetaDataVocabulary.wasGeneratedBy.asNode(), nodeResultGraph));
-        lstTriples.add(new Triple(nodeResultGraph, MetaDataVocabulary.uriName.asNode(), NodeFactory.createURI(crawlingActivity.getCrawleableUri().getUri().toString())));
-        //TODO lstTriples.add(new Triple(nodeCrawlingActivity, NodeFactory.createURI("sq:hostedOn"), NodeFactory.createLiteral(datasetPrefix)));
-        //TODO for Meher: Merge new change from other branch manually for hadPlan
-        //lstTriples.add(new Triple(nodeCrawlingActivity,NodeFactory.createURI("prov:hadPlan"),NodeFactory.createLiteral(crawlingActivity.getHadPlan())));
+        Model model  = ModelFactory.createDefaultModel();
+        CrawleableUri uri = crawlingActivity.getCrawleableUri();
+
+        Node nodeResultGraph = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Result_"+ crawlingActivity.getGraphId(uri))));
+        Node nodeCrawlingActivity = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl(crawlingActivity.getId())));
+        Node Association = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Worker-checking-Crawl-guide")));
+        Node crawling = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Crawl-guide")));
+
+        lstTriples.add(new Triple(nodeCrawlingActivity,RDF.type.asNode(),Prov.Activity.asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Prov.startedAtTime.asNode(), model.createTypedLiteral(crawlingActivity.getLocalDateTime(),XSDDatatype.XSDdateTime).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Prov.endedAtTime.asNode(), model.createTypedLiteral(crawlingActivity.getLocalDateTime(),XSDDatatype.XSDdateTime).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Sq.status.asNode(), NodeFactory.createLiteral(crawlingActivity.getState().toString())));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Sq.numberOfTriples.asNode(), model.createTypedLiteral(crawlingActivity.getNumTriples(),XSDDatatype.XSDint).asNode()));
```

Comment:
Using `xsd:int` is not a good idea since some datasets may have a large amount of triples while `xsd:int` can only represent 2^31 as maximum positive number. Please check https://www.w3.org/TR/xmlschema11-2/ for better alternatives.